src/mds/MDCache.cc

```diff
@@ -12217,6 +12217,15 @@ void MDCache::enqueue_scrub(
   }
 
   C_MDS_EnqueueScrub *cs = new C_MDS_EnqueueScrub(f, fin);
+  if (tag.empty()) {
+    uuid_d uuid_gen;
+    uuid_gen.generate_random();
+
+    char uuid_str[37];
+    uuid_gen.print(uuid_str);
+    tag = uuid_str;
+  }
+
```

Comment:
non-empty scrub tag will get stored in inodes' xattr.  This change will introduce unnecessary writes

Reply:
the tag will be persisted for every inode in the subtree being scrubbed, right? we could skip this IO operation if the tag is internal and not provided via CLI.

Reply:
yes, current code store tag for every inodes.  yes, we could modify the code to not store internal tag